File: documentation-examples-java/src/java/dev/typr/foundations/docs/duckdb/NestedCollections.java
package dev.typr.foundations.docs.duckdb;

import dev.typr.foundations.DuckDbType;
import dev.typr.foundations.DuckDbTypes;
import java.util.List;

@SuppressWarnings("unused")
public class NestedCollections {
  // start
  // Nested collections compose — any combination is legal.
  DuckDbType<List<List<Integer>>> grid = DuckDbTypes.integer.list().list();
  DuckDbType<List<List<Float>>> matrix = DuckDbTypes.float_.array(3).array(3);
  DuckDbType<List<List<String>>> variableRows = DuckDbTypes.varchar.list().array(4);
  // stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/duckdb/NestedCollections.kt
package dev.typr.foundationskt.docs.duckdb

import dev.typr.foundationskt.*

@Suppress("unused")
class NestedCollections {
    //start
    // Nested collections compose — any combination is legal.
    val grid: DuckDbType<List<List<Int>>> = DuckDbTypes.integer.list().list()
    val matrix: DuckDbType<List<List<Float>>> = DuckDbTypes.float_.array(3).array(3)
    val variableRows: DuckDbType<List<List<String>>> = DuckDbTypes.varchar.list().array(4)
    //stop
}
